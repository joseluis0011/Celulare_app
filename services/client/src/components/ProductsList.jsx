import React from 'react';


const ProductsList = (props) => {
  return (
    <div>
      <table className="table is-narrow is-striped is-fullwidth is-hoverable">
      <thead>
        <tr>
          <th>Nombre</th>
          <th>Cantidad</th>
          <th>Serie</th>
          <th>Modelo</th>
          <th>Marca</th>
        </tr>
      </thead>
      <tbody>
      {
        props.products.map((product) => {
          return (
            <tr key={product.id}>
            <td>{ product.nombre}</td>
            <td>{ product.cantidad}</td>
            <td>{ product.serie}</td>
            <td>{ product.modelo}</td>
            <td>{ product.marca}</td>
            </tr>
          )
        })
      }
      </tbody>
      </table>
    </div>
  )
};


export default ProductsList;