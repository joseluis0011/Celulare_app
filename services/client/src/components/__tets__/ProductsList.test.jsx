import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
 
import ProductsList from '../ProductsList';
 
const products = [
  { 
    'marca': 'xiomi',
    'modelo': 'S4',
    'serie': 'MK3526',
    'cantidad': 12,
    'id': 1,
    'nombre': 'Sansung'
  },
  { 
    'marca': 'apple',
    'modelo': 'M5',
    'serie': 'JJ3526',
    'cantidad': 12,
    'id': 2,
    'nombre': 'ipone'
  }
];
 
test('ProductsList renders properly', () => {
  const wrapper = shallow(<ProductsList products={products}/>);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('Oreo');
});

test('ProductsList renders a snapshot properly', () => {
  const tree = renderer.create(<ProductsList products={products}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
