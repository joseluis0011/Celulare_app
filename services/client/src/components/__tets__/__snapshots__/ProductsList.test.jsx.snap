// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList renders a snapshot properly 1`] = `
<div>
  <table
    className="table is-narrow is-striped is-fullwidth is-hoverable"
  >
    <thead>
      <tr>
        <th>
          Nombre
        </th>
        <th>
          Cantidad
        </th>
        <th>
          Serie
        </th>
        <th>
          Modelo
        </th>
        <th>
          Marca
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Sansung
        </td>
        <td>
          12
        </td>
        <td>
          MK3526
        </td>
        <td>
          S4
        </td>
        <td>
          xiomi
        </td>
      </tr>
      <tr>
        <td>
          ipone
        </td>
        <td>
          12
        </td>
        <td>
          JJ3526
        </td>
        <td>
          M5
        </td>
        <td>
          apple
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
